statsd/worker_test.go

```diff
@@ -0,0 +1,35 @@
+package statsd
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestShouldSample(t *testing.T) {
+	rates := []float64{0.01, 0.05, 0.1, 0.25, 0.5, 0.75, 0.9, 0.99, 1.0}
+	iterations := 50_000
+	worker := newWorker(newBufferPool(1, 1, 1), nil)
+	for _, rate := range rates {
+		rate := rate // Capture range variable.
```

Comment:
I don't think we need this since `t.Run` will block until the  test is done.

Reply:
Ah, you're correct, that makes me realize I'd intended to add `t.Parallel()` to run each subtest in parallel. I'll make that change and push it.